lib/Model.js

```diff
@@ -365,99 +376,101 @@ Model.create = function(NewModel, obj, options, next) {
+    if(!validKeyValue(key[hashKeyName])) {
+      let keyVal = key;
+      key = {};
+      key[hashKeyName] = keyVal;
+    }
 
+    if(schema.rangeKey && !validKeyValue(key[schema.rangeKey.name])) {
+      reject( new errors.ModelError('Range key required: ' + schema.rangeKey.name) );
+    }
 
-  let getReq = {
-    TableName: NewModel.$__.name,
-    Key: {}
-  };
 
-  getReq.Key[hashKeyName] = schema.hashKey.toDynamo(key[hashKeyName], undefined, key);
+    let getReq = {
+      TableName: NewModel.$__.name,
+      Key: {}
+    };
 
-  if(schema.rangeKey) {
-    let rangeKeyName = schema.rangeKey.name;
-    getReq.Key[rangeKeyName] = schema.rangeKey.toDynamo(key[rangeKeyName], undefined, key);
-  }
+    getReq.Key[hashKeyName] = schema.hashKey.toDynamo(key[hashKeyName], undefined, key);
 
-  if(options.attributes) {
-    getReq.AttributesToGet = options.attributes;
-  }
+    if(schema.rangeKey) {
+      let rangeKeyName = schema.rangeKey.name;
+      getReq.Key[rangeKeyName] = schema.rangeKey.toDynamo(key[rangeKeyName], undefined, key);
+    }
 
-  if(options.consistent) {
-    getReq.ConsistentRead = true;
-  }
+    if(options.attributes) {
+      getReq.AttributesToGet = options.attributes;
+    }
 
-  let newModel$ = NewModel.$__;
+    if(options.consistent) {
+      getReq.ConsistentRead = true;
+    }
 
-  function get () {
-    debug('getItem', getReq);
-    newModel$.base.ddb().getItem(getReq, function(err, data) {
-      if(err) {
-        debug('Error returned by getItem', err);
-        return deferred.reject(err);
-      }
+    let newModel$ = NewModel.$__;
 
-      debug('getItem response', data);
+    function get () {
+      debug('getItem', getReq);
+      newModel$.base.ddb().getItem(getReq, function(err, data) {
+        if(err) {
+          debug('Error returned by getItem', err);
+          return reject(err);
+        }
 
-      if(!Object.keys(data).length) {
-        return deferred.resolve();
-      }
+        debug('getItem response', data);
 
-      let model = new NewModel();
+        if(!Object.keys(data).length) {
+          return resolve();
+        }
 
-      model.$__.isNew = false;
-      try {
-        schema.parseDynamo(model, data.Item);
-      } catch(e){
-        debug('cannot parse data', data);
-        return deferred.reject(e);
-      }
+        let model = new NewModel();
 
-      if (schema.expires && schema.expires.returnExpiredItems === false && model[schema.expires.attribute] < new Date()) {
-      	deferred.resolve();
-      }
+        model.$__.isNew = false;
+        try {
+          schema.parseDynamo(model, data.Item);
+        } catch(e){
+          debug('cannot parse data', data);
+          return reject(e);
+        }
 
-      debug('getItem parsed model', model);
-      deferred.resolve(model);
-    });
-  }
+        if (schema.expires && schema.expires.returnExpiredItems === false && model[schema.expires.attribute] < new Date()) {
+          resolve();
```

Comment:
It looks like this was a carbon copy of the old version. Which is great. But I wonder if this is a bug. Shouldn't this be the following?

```suggestion
          return resolve();
```

I could be totally wrong tho... what do you think?